src/arithmetic/arithmetic_expression.c

```diff
@@ -182,17 +195,28 @@ bool AR_EXP_ReduceToScalar(AR_ExpNode **root) {
 			if(!reduce_children) return false;
 
 			// All child nodes are constants, make sure function is marked as reducible.
-			AR_FuncDesc *func_desc = AR_GetFunc((*root)->op.func_name);
+			AR_FuncDesc *func_desc = AR_GetFunc(root->op.func_name);
 			assert(func_desc);
 			if(!func_desc->reducible) return false;
 
 			// Evaluate function.
-			SIValue v = AR_EXP_Evaluate(*root, NULL);
+			SIValue v = AR_EXP_Evaluate(root, NULL);
 			if(SIValue_IsNull(v)) return false;
 
 			// Reduce.
-			AR_EXP_Free(*root);
-			*root = AR_EXP_NewConstOperandNode(v);
+			// Clear children and function context.
+			for(int child_idx = 0; child_idx < root->op.child_count; child_idx++) {
+				AR_EXP_Free(root->op.children[child_idx]);
+			}
+			rm_free(root->op.children);
+			if(root->op.type == AR_OP_AGGREGATE) {
+				AggCtx_Free(root->op.agg_func);
+			}
```

Comment:
Don't duplicate `AR_EXP_Free`, introduce `_AR_EXP_Free` which frees the internals,
have `AR_EXP_Free` call it.